// @hubspot/ui-extensions/dist/crm/components.js
import { createExtensionComponent } from '../utils/createExtensionComponent';
const CrmObjectProperty = createExtensionComponent('CrmObjectProperty');
const CrmPropertyList = createExtensionComponent('CrmPropertyList');
const CrmAssociationTable = createExtensionComponent('CrmAssociationTable');
const CrmDataHighlight = createExtensionComponent('CrmDataHighlight');
const CrmReport = createExtensionComponent('CrmReport');
const CrmAssociationPivot = createExtensionComponent('CrmAssociationPivot');
const CrmAssociationPropertyList = createExtensionComponent('CrmAssociationPropertyList');
const CrmStageTracker = createExtensionComponent('CrmStageTracker');
const CrmActionButton = createExtensionComponent('CrmActionButton');
const CrmActionLink = createExtensionComponent('CrmActionLink');
const CrmCardActions = createExtensionComponent('CrmCardActions');
export { CrmObjectProperty, CrmPropertyList, CrmAssociationTable, CrmDataHighlight, CrmReport, CrmAssociationPivot, CrmAssociationPropertyList, CrmStageTracker, CrmActionButton, CrmActionLink, CrmCardActions, };
