// @hubspot/api-client/lib/codegen/crm/associations/v4/types/ObjectParamAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectDefinitionsApi = exports.ObjectBatchApi = void 0;
const ObservableAPI_1 = require("./ObservableAPI");
class ObjectBatchApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_1.ObservableBatchApi(configuration, requestFactory, responseProcessor);
    }
    archive(param, options) {
        return this.api.archive(param.fromObjectType, param.toObjectType, param.batchInputPublicAssociationMultiArchive, options).toPromise();
    }
    archiveLabels(param, options) {
        return this.api.archiveLabels(param.fromObjectType, param.toObjectType, param.batchInputPublicAssociationMultiPost, options).toPromise();
    }
    create(param, options) {
        return this.api.create(param.fromObjectType, param.toObjectType, param.batchInputPublicAssociationMultiPost, options).toPromise();
    }
    createDefault(param, options) {
        return this.api.createDefault(param.fromObjectType, param.toObjectType, param.batchInputPublicDefaultAssociationMultiPost, options).toPromise();
    }
    getPage(param, options) {
        return this.api.getPage(param.fromObjectType, param.toObjectType, param.batchInputPublicFetchAssociationsBatchRequest, options).toPromise();
    }
}
exports.ObjectBatchApi = ObjectBatchApi;
const ObservableAPI_2 = require("./ObservableAPI");
class ObjectDefinitionsApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_2.ObservableDefinitionsApi(configuration, requestFactory, responseProcessor);
    }
    archive(param, options) {
        return this.api.archive(param.fromObjectType, param.toObjectType, param.associationTypeId, options).toPromise();
    }
    create(param, options) {
        return this.api.create(param.fromObjectType, param.toObjectType, param.publicAssociationDefinitionCreateRequest, options).toPromise();
    }
    getAll(param, options) {
        return this.api.getAll(param.fromObjectType, param.toObjectType, options).toPromise();
    }
    update(param, options) {
        return this.api.update(param.fromObjectType, param.toObjectType, param.publicAssociationDefinitionUpdateRequest, options).toPromise();
    }
}
exports.ObjectDefinitionsApi = ObjectDefinitionsApi;
